File: problem_solving_algorithms/easy/compare_the_triplets.py
# Compare the Triplets
# https://www.hackerrank.com/challenges/compare-the-triplets/problem

def compare(a, b):
    lis_a = list(map(int, a.split()))
    lis_b = list(map(int, b.split()))
    score = [0, 0]

    for i in range(0, len(lis_a)):
        if lis_a[i] > lis_b[i]:
            score[0] += 1
        elif lis_a[i] < lis_b[i]:
            score[1] += 1
        else:
            pass

    print(" ".join(map(str, score)))

a = "17 28 30"
b = "99 16 8"
compare(a, b)

File: problem_solving_algorithms/easy/compare_the_triplets.js
// Compare the Triplets
// https://www.hackerrank.com/challenges/compare-the-triplets/problem

// function compareTriplets(x, y) {
//     var a = x.split(' ');
//     var b = y.split(' ');

//     var a_arr = [];
//     var b_arr = [];

//     for (var i = 0; i < a.length; i++) {
//         a_arr.push(parseInt(a[i]));
//         b_arr.push(parseInt(b[i]));
//     }

//     var aScore = 0;
//     var bScore = 0;

//     for (var n = 0; n < a_arr.length; n++) {
//         if (a_arr[n] > b_arr[n]) {
//             aScore++;
//         } else if (a_arr[n] < b_arr[n]) {
//             bScore++;
//         }
//     }

//     return aScore.toString() + " " + bScore.toString();
// }


// Accepted Solution
function compareTriplets(a, b) {
    var result = [];
    var aScore = 0, bScore = 0;

    for (var i = 0; i < a.length; i++) {
        if (a[i] > b[i]) aScore++;
        if (a[i] < b[i]) bScore++;
    }
    result[0] = aScore;
    result[1] = bScore;
    return result;
}

console.log(compareTriplets('5 6 7', '3 6 10'));